### PR TITLE
geodesic distance

### PR DESCRIPTION
**Pull request by michaelkirk**
Integrates with the geographiclib rust port started by @Psykopear and touched up by me. 

FIXES https://github.com/georust/geo/issues/234
FIXES https://github.com/georust/geo/issues/425

### CODE REVIEW

File: geo/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,8 @@ travis-ci = { repository = "georust/geo" }
 num-traits = "0.2"
 serde = { version = "1.0", optional = true, features = ["derive"] }
 rstar = { version = "0.7" }
+geographiclib-rs = { version = "0.1", git = "https://github.com/michaelkirk/geographiclib-rs" }
```
Comment (michaelkirk) [inline geo/Cargo.toml:21]: I don't want to merge this repo reference as is. 

My preference would be moving the geographiclib-rs repo into the georust org and publishing from there. Thoughts?

File: geo/src/algorithm/geodesic_distance.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ///     5_585_234., // meters
+    ///     distance.round()
+    /// );
+    /// ```
+    /// [Karney (2013)]:  https://arxiv.org/pdf/1109.4448.pdf
+    fn geodesic_distance(&self, rhs: &Rhs) -> T;
+}
+
+impl GeodesicDistance<f64> for Point<f64> {
+    fn geodesic_distance(&self, rhs: &Point<f64>) -> f64 {
```
Comment (michaelkirk) [inline geo/src/algorithm/geodesic_distance.rs:43]: Note that, currently anyway, geographiclib only works with f64.

File: geo/benches/geodesic_distance.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+use geo::prelude::*;
+
+fn criterion_benchmark(c: &mut criterion::Criterion) {
+    c.bench_function("geodesic distance f64", |bencher| {
+        let a = geo::Point::<f64>::new(17.107558, 48.148636);
+        let b = geo::Point::<f64>::new(16.372477, 48.208810);
+
+        bencher.iter(|| {
+            let _ = a.geodesic_distance(&b);
```
Comment (michaelkirk) [inline geo/benches/geodesic_distance.rs:13]: The geodesic computation is 2-3x slower than vincenty

```
geodesic distance f64   time:   [873.59 ns 896.28 ns 922.24 ns]                                   
vincenty distance f64   time:   [357.93 ns 359.94 ns 361.92 ns]                                  
```

File: geo/Cargo.toml (hunk 1 above)
Comment (michaelkirk) [reply geo/Cargo.toml:21]: /cc @Psykopear

File: geo/benches/geodesic_distance.rs (hunk 3 above)
Comment (michaelkirk) [reply geo/benches/geodesic_distance.rs:13]: FYI we can get this down 20% if we're willing to make a hard dependency on the `lazy_static` crate (we already have several transitive dependencies on it). Shown here:  https://github.com/michaelkirk/rust-geo/commit/7e2c22c91b019f032afc9d23b9858f83b6b01b1e

geodesic distance f64   time:   [720.97 ns 725.92 ns 732.69 ns]

File: geo/Cargo.toml (hunk 1 above)
Comment (frewsxcv) [reply geo/Cargo.toml:21]: if you get @Psykopear's approval, +1 to move into the georust org. let me know if you need permissions for that

File: geo/benches/geodesic_distance.rs (hunk 3 above)
Comment (frewsxcv) [reply geo/benches/geodesic_distance.rs:13]: would `lazy_static` be added to this crate or the `geodesiclib` crate? + 1 from me either way

Comment (michaelkirk) [reply geo/benches/geodesic_distance.rs:13]: I was originally thinking here in geo, but now that you mention it, it makes more sense in the geographiclib crate so any other potential users could benefit.

Comment (michaelkirk) [reply geo/benches/geodesic_distance.rs:13]: Done in https://github.com/michaelkirk/geographiclib-rs/pull/4

File: geo/Cargo.toml (hunk 1 above)
Comment (Psykopear) [reply geo/Cargo.toml:21]: Hi @frewsxcv and @michaelkirk 
The code is released under MIT License on purpose, so please take whatever you need and use it how you please.
I just had an email exchange with Michael, with the work he did you can consider him an author as much as me. As I said to him I probably won't be working on this for a while.

**Comment by michaelkirk** [comment]
bors r=frewsxcv

**Pull request merged**